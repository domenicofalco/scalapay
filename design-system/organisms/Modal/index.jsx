import PropTypes from "prop-types";
import Portal from "molecules/Portal";
import Button from "atoms/Button";
import { white } from "colors";
import styled from "styled-components";

const Wrapper = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
  background: rgba(0, 0, 0, 0.5);
  z-index: 9999;
`;

const Container = styled.div`
  width: 90%;
  max-width: 450px;
  margin: 0;
  position: absolute;
  border-radius: 20px;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background: ${white};
`;

const BtnWrap = styled.span`
  position: absolute;
  right: 10px;
  top: 10px;
`;

const ChildWrap = styled.div`
  padding: 20px 40px 20px 20px;
`;

const Modal = ({ showModal, children, onClose }) => {
  if (!showModal) return null;

  return (
    <Portal selector="#modal-root">
      <Wrapper>
        <Container>
          <BtnWrap>
            <Button type="button" onClick={onClose} variant="roundIcon">
              X
            </Button>
          </BtnWrap>
          <ChildWrap>{children}</ChildWrap>
        </Container>
      </Wrapper>
    </Portal>
  );
};

Modal.propTypes = {
  showModal: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node
};

export default Modal;
