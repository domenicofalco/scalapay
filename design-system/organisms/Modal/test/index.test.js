import React from "react";
import { cleanup, render } from "@testing-library/react";
import Modal from "../";

jest.mock("molecules/Portal", () => {
  return props => <div {...props}>{props.children}</div>;
});

const props = {
  showModal: true,
  children: <div>a children</div>,
  onClose: () => null
};

afterEach(cleanup);

describe("organisms => Modal", () => {
  it("It should match snapshots and render children", () => {
    const { asFragment, getByText } = render(
      <Modal {...props}>
        <div>Modal</div>
      </Modal>
    );

    const children = getByText(/Modal/i);
    expect(children).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it("It should not render the modal", () => {
    const { container } = render(
      <Modal {...props} showModal={false}>
        <div className="a_children">Modal</div>
      </Modal>
    );

    const children = container.querySelector(".a_children");
    expect(children).toBeFalsy();
  });
});
