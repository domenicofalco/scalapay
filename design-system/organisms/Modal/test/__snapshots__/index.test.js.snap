// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms => Modal It should match snapshots and render children 1`] = `
<DocumentFragment>
  <div
    selector="#modal-root"
  >
    <div
      class="Modal__Wrapper-sc-1dpt69b-0 hKXsaP"
    >
      <div
        class="Modal__Container-sc-1dpt69b-1 bnQqjN"
      >
        <span
          class="Modal__BtnWrap-sc-1dpt69b-2 cHauyq"
        >
          <button
            class="Button__roundIcon-sc-6egjvv-2 feZvzI"
            data-variant="roundIcon"
            type="button"
          >
            X
          </button>
        </span>
        <div
          class="Modal__ChildWrap-sc-1dpt69b-3 kVnOOr"
        >
          <div>
            Modal
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
